File: server/src/Services/NotificationService/NotificationService.API/Entities/Notification.cs
namespace NotificationService.API.Entities;

public class Notification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; } = Guid.Empty; // Empty is admin's notification
    public NotificationType Type { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public Dictionary<string, object>? Data { get; set; }
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public bool IsSeen { get; set; } = false;
}

File: client/mobile/vousphere/lib/data/models/Notification.dart
class NotificationModel {
  final String id;
  final String userId;
  final String type;
  final String title;
  final String message;
  final Map<String, dynamic> data;
  final DateTime createdAt;
  final bool isSeen;

  NotificationModel({
    required this.id,
    required this.userId,
    required this.type,
    required this.title,
    required this.message,
    required this.data,
    required this.createdAt,
    required this.isSeen,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      id: json['id'],
      userId: json['userId'],
      type: json['type'],
      title: json['title'],
      message: json['message'],
      data: json['data'],
      createdAt: DateTime.parse(json['createdAt']),
      isSeen: json['isSeen'],
    );
  }
}
